src/form/Input/InlineInput/InlineInput.tsx

```diff
@@ -40,22 +51,54 @@ export interface InlineInputProps
 export const InlineInput: FC<InlineInputProps> = forwardRef(
   (
     {
+      className,
       inputClassName,
-      placeholderIsMinWidth = true,
+      placeholder,
+      placeholderIsMinWidth,
       theme: customTheme,
+      extraWidth,
+      onChange,
       ...rest
     },
-    ref: Ref<HTMLInputElement>
+    ref: RefObject<InputRef>
   ) => {
     const theme: InputTheme = useComponentTheme('input', customTheme);
+    const spanStyles = 'relative opacity-0 whitespace-pre select-none';
+    const [inputValue, setInputValue] = useState('');
+    const [placeholderValue, setPlaceholderValue] = useState(placeholder);
+
+    const onInputValueChange = event => {
+      setInputValue(event.target.value);
+      onChange(event);
+    };
+
+    const setRef = (node: HTMLInputElement) => {
+      ref?.current?.inputRef;
+
+      const placeholder = node?.placeholder;
+      const value = node?.value;
+
+      placeholder && setPlaceholderValue(placeholder);
+      value && setInputValue(value);
+    };
```

Comment:
is this a callback ref? Cool, I'm not familiar never used it. What's line 76 `ref?.current?.inputRef` doin?

If you pass a `ref` to `InlineInput` it doesn't get attached the actual input anymore right? So it's not behaving as you'd expect